keps/sig-apps/961-maxunavailable-for-statefulset/README.md

```diff
@@ -291,80 +699,201 @@ No, the default behavior remains the same.
+- Cluster Stability Metrics: Key indicators of cluster health, such as increased error rates in control plane components or higher pod restart rates, post-feature adoption, can also guide the decision to rollback.
 
 ###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
-Will be tested when graduating to Beta.
+
+A manual test was performed, as follows:
+
+1. Create a cluster in 1.23.
+2. Upgrade to 1.24.
+3. Create StatefulSet A with spec.updateStrategy.rollingUpdate.maxUnavailable set to 3, with 6 replicas
+4. Verify a rollout and check if only 3 pods are unavailable at a time ([currently with a bug if podManagementPolicy is set to Parallel](https://github.com/kubernetes/kubernetes/issues/112307))
+5. Downgrade to 1.23.
+6. Verify that the rollout only has 1 pod unavailable at a time, similar to setting maxUnavailable to 1
+7. Create another StatefulSet B not setting maxUnavailable (leaving it nil)
+8. Upgrade to 1.24.
+9. Verify that the rollout has default behavior of only having one pod unavailable at a time
+   Verify that the `maxUnavailable` can be set again to StatefulSet A and test the rollout behavior
+
+TODO:
+A manual test will be performed, as follows:
+
+1. Create a cluster in 1.29.
+2. Upgrade to 1.30.
+3. Create StatefulSet A with spec.updateStrategy.rollingUpdate.maxUnavailable set to 3, with 6 replicas
+4. Verify a rollout and check if only 3 pods are unavailable at a time
+5. Check if rollout is also fine with podManagementPolicy set to Parallel
+4. Downgrade to 1.29.
+6. Verify that the rollout only has 1 pod unavailable at a time, similar to setting maxUnavailable to 1 (MaxUnavailableStatefulSet feature gate disabled by default).
+7. Create another StatefulSet B not setting maxUnavailable (leaving it nil)
+8. Upgrade to 1.30.
+9. Verify that the rollout has default behavior of only having one pod unavailable at a time
+   Verify that the `maxUnavailable` can be set again to StatefulSet A and test the rollout behavior
 
 
 ###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
 No
 
 ### Monitoring Requirements
 
 ###### How can an operator determine if the feature is in use by workloads?
+
 If their StatefulSet rollingUpdate section has the field maxUnavailable specified with
-a value different than 1.
-The below command should show maxUnavailable value:
+a value different than 1. While in alpha and beta, the feature-gate needs to be enabled.
+
+The command bellow should show the maxUnavailable value:
 ```
 kubectl get statefulsets -o yaml | grep maxUnavailable
 ```
 
 ###### How can someone using this feature know that it is working for their instance?
-TODO when we reach Beta
+
+<!-- need to decide if I can use statefulset_unavailability_violation here -->
```

Comment:
Could no think of something less manual if we drop statefulset_unavailability_violation. If we decide to keep it we can use it here. Also open to suggestions

Reply:
✅

Reply:
Added with some context in some parts of the text